### PR TITLE
[test] Added Magic Button triggering some events every 5 seconds

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added Magic Button triggering some events every 5 seconds
- Minor improvements

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.test.magic/src/main/resources/OH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -11,6 +11,17 @@
...
 	</thing-type>
 
+	<thing-type id="button">
+		<label>Magic Button</label>
+		<description>A button triggering different events.</description>
+
+		<channels>
+			<channel typeId="system.rawbutton" id="rawbutton"/>
+			<channel typeId="system.button" id="button"/>
+			<channel typeId="system.rawrocker" id="rawrocker"/>
```
Comment (wborn) [inline bundles/org.openhab.core.test.magic/src/main/resources/OH-INF/thing/thing-types.xml:21]:
```suggestion
			<channel typeId="system.button" id="button"/>
			<channel typeId="system.rawrocker" id="rawrocker"/>
```

**Review by wborn — CHANGES_REQUESTED** [review]
I think the audience will like these new magic tricks. But there is a small issue in the added magic. :wink:

File: bundles/org.openhab.core.test.magic/src/main/resources/OH-INF/thing/thing-types.xml (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.openhab.core.test.magic/src/main/resources/OH-INF/thing/thing-types.xml:21]: Apparently we have all this different buttons.i the default system channel types:

https://github.com/openhab/openhab-core/blob/f9d201c35bacfe48270526419b8ac91bf488db30/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java#L67-L69

Comment (wborn) [reply bundles/org.openhab.core.test.magic/src/main/resources/OH-INF/thing/thing-types.xml:21]: It was still a bit early when I reviewed this so I thought it was using the same channel type twice. But I now see they are different. 🕵️‍♂️

**Review by wborn — APPROVED** [review]
Thanks!

**Pull request merged**